### PR TITLE
Use per package global lock for container uploads instead of memory lock

### PR DESCRIPTION
**Pull request by lunny**
This PR uses a per-package lock instead of a global lock for container uploads. So that different packages can be uploaded simultaneously.

### CODE REVIEW

File: routers/api/packages/container/blob.go [hunk 1]
```diff
@@ -95,7 +95,8 @@ func getOrCreateUploadVersion(ctx context.Context, pi *packages_service.PackageI
 
 	// FIXME: Replace usage of mutex with database transaction
 	// https://github.com/go-gitea/gitea/pull/21862
-	uploadVersionMutex.Lock()
+	uploadVersionMutex.Start(strings.ToLower(pi.Name))
```
Comment (wxiaoguang) [inline routers/api/packages/container/blob.go]: Wrong

Comment (lunny) [reply routers/api/packages/container/blob.go]: 96ec57c3e58aa6f8db873024dfd8c6155fe1680a

**Comment by wolfogre** [comment]
Wait for #31908?

**Comment by lunny** [comment]
Depends on #31813

**Review by wolfogre — APPROVED** [review]

**Comment by lunny** [comment]
The lock key should contain owner id.

**Comment by yp05327** [comment]
Do we need to add lock in `mountBlob`?

**Review by yp05327 — APPROVED** [review]

**Pull request merged**